### ISSUE TITLE
Split API functionality into a separate library

### ISSUE THREAD

**Issue by dgomes** [pypolestar/polestar_api#14]
In order for this integration to become an official HA integration all API call must be moved into a separate python library published on [pypi](https://pypi.org) 

I propose the creation of library `pypolestar` in a separate repository and that best practices such as those in  https://github.com/pyenphase/pyenphase are followed

**Comment by jschlyter** [comment]
I agree. I'd be happy to help setting up the infrastructure if needed.

**Comment by dgomes** [comment]
@leeyuentuen I think there are already a couple devs available to work on this, would you care to create a repository or delegate in one of us the task ? maybe creating a GitHub organisation to manage the library + custom_integration ?

**Comment by leeyuentuen** [comment]
i need to find it out how that stuff works.
atm create a GitHub pypolestar: https://github.com/leeyuentuen/pypolestar
on PyPI I've created a pending project:
![image](https://github.com/leeyuentuen/polestar_api/assets/1487966/a7ba5985-c53c-4fcb-857b-9411b6c85b36)

**Comment by dgomes** [comment]
things are looking real good in the current custom_integration.

Pushing the release of the separate library would enable the work in an initial PR to HA core with limited functionality (we don't want to feature creep reviewers, a simple PR will get better chances of being reviewed quicker)

**Comment by jschlyter** [comment]
Splitting to a separate library is pending on https://github.com/pypolestar/pypolestar/issues/10, waiting on action from @leeyuentuen (nudge, nudge)

**Comment by jschlyter** [comment]
https://github.com/pypolestar/pypolestar now up to date, will do some cleanup before publishing a new version

**Comment by jschlyter** [comment]
https://github.com/pypolestar/pypolestar now updated on PyPI.
